webapp/src/projects.tsx

```diff
@@ -570,7 +570,6 @@ export class ProjectsCarousel extends data.Component<ProjectsCarouselProps, Proj
         this.closeDetail = this.closeDetail.bind(this);
         this.closeDetailOnEscape = this.closeDetailOnEscape.bind(this);
         this.reload = this.reload.bind(this);
-        this.newProject = this.newProject.bind(this);
```

Comment:
Is the reason we no longer need this because of the change in how the function get scalled in `onClick` below? This kind of binding is still a bit of a mystery to me...

Reply:
Yes -- the issue with passing this.newProject is that javascript basically loses the 'this` context for explicit methods when passing (here's an article on it / the weirdness https://www.freecodecamp.org/news/this-is-why-we-need-to-bind-event-handlers-in-class-components-in-react-f7ea1a6f93eb/) -- for what it's worth, my preference is just to switch to arrow function (`newProject = () => {`), which does a similar job. Or, well, using functional components in new code, which sidesteps this issue completely.

Reply:
Thanks for the info! Very helpful article.